api/context/rest_client.go

```diff
@@ -0,0 +1,105 @@
+package context
+
+import (
+	"fmt"
+
+	"github.com/CircleCI-Public/circleci-cli/api/rest"
+)
+
+type restClient struct {
+	client *rest.Client
+
+	orgID   *string
+	vcsType *string
+	orgName *string
+}
+
+func toSlug(vcs, org string) *string {
+	slug := fmt.Sprintf("%s/%s", vcs, org)
+	return &slug
+}
+
+// createListContextsParams is a helper to create ListContextsParams from the content of the ContextRestClient
+func (c *restClient) createListContextsParams() (*ListContextsWithRestParams, error) {
```

Comment:
Why return a pointer, here?

Reply:
I think, as a Go practice I like to return a unusable value in case of error, returning a pointer here allow to return `nil` in case of error.

Reply:
That's what the zero value is for. Returning `nil` is just returning the zero value of a pointer type.

For example, [here](https://go.dev/doc/tutorial/handle-errors), in the Go tutorial, the func `return "", errors.New()`. Other examples [here](https://github.com/golang/go/blob/master/src/net/tcpsock.go#L157-L159) and [here](https://github.com/golang/go/blob/master/src/strconv/atoi.go#L81-L83) from the Go sources.